grove/alpha/bernstein_vazirani/bernstein_vazirani.py

```diff
@@ -1,171 +1,203 @@
-
-    where :math:`\\text{xor}` is taken bitwise.
-
-    Allocates one scratch bit.
-
-    :param list(int) vec_a: a vector of length :math:`n`
-                                containing only ones and zeros.
-                                The order is taken to be
-                                most to least significant bit.
-    :param int b: a 0 or 1
-    :param list(int) qubits: List of qubits that enter as input
-                             :math:`\\vert x\\rangle`.
-                             Must be the same length (:math:`n`)
-                             as :math:`\\mathbf{a}`
-    :param int ancilla: Ancillary qubit to serve as input
-                        :math:`\\vert y\\rangle`,
-                        where the answer will be written to.
-    :return: A program that performs the above unitary transformation.
-    :rtype: Program
-    """
-    assert len(vec_a) == len(qubits), \
-        "vec_a must be the same length as the number of input qubits"
-    assert all(list(map(lambda x: x in {0, 1}, vec_a))), \
-        "vec_a must be a list of 0s and 1s"
-    assert b in {0, 1}, "b must be a 0 or 1"
-
-    n = len(qubits)
-    p = pq.Program()
-    if b == 1:
-        p.inst(X(ancilla))
-    for i in range(n):
-        if vec_a[i] == 1:
-            p.inst(CNOT(qubits[n - 1 - i], ancilla))
-    return p
-
-
-def bernstein_vazirani(oracle, qubits, ancilla):
+           (\\mathbf{a}\\in\\{0,1\\}^n, b\\in\\{0,1\\})
+    :param String a: a string of 0's and 1's that represents the dot-product partner in :math:`f`
+    :param String b: 0 or 1 as a string representing the bias term in :math:`f`
+    :return: A dictionary containing all possible bitstring of length equal to :math:`a` and the
+    function value :math:`f`
+    :rtype: Dict[String, String]
     """
-    Implementation of the Bernstein-Vazirani Algorithm.
-
-    Given a list of input qubits and an ancilla bit,
-    all initially in the :math:`\\vert 0\\rangle` state,
-    create a program that can find :math:`\\vec{a}`
-    with one query to the given oracle.
-
-    :param Program oracle: Program representing unitary application of function
-    :param list(int) qubits: List of qubits that enter as state
-                             :math:`\\vert x\\rangle`.
-    :param int ancilla: Ancillary qubit to serve as input
-                        :math:`\\vert y\\rangle`,
-                        where the answer will be written to.
-    :return: A program corresponding to the desired instance of the
-             Bernstein-Vazirani Algorithm.
-    :rtype: Program
-    """
-    p = pq.Program()
-
-    # Put ancilla bit into minus state
-    p.inst(X(ancilla), H(ancilla))
+    n_bits = len(a)
+    bit_map = {}
+    for bit_val in range(2 ** n_bits):
+        bit_map[np.binary_repr(bit_val, width=n_bits)] = \
+            str((int(utils.bitwise_dot_product(
+                np.binary_repr(bit_val, width=n_bits), a)) + int(b, 2)) % 2)
```

Comment:
Can we get rid of the backslash?:
PEP8:
"The preferred way of wrapping long lines is by using Python's implied line continuation inside parentheses, brackets and braces. Long lines can be broken over multiple lines by wrapping expressions in parentheses. These should be used in preference to using a backslash for line continuation."

Reply:
done